clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -588,13 +587,13 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
                     TopicPartition tp = entry.getKey();
                     ProduceResponse.PartitionResponse partResp = entry.getValue();
                     ProducerBatch batch = batches.get(tp);
-                    completeBatch(batch, partResp, correlationId, now, receivedTimeMs + produceResponse.throttleTimeMs());
```

Comment:
Is the idea that we can remove the throttling here, and rely on the existing NetworkClient throttle instead?

Reply:
Yes. This logic was only active for max.inflight=1 so it's confusing which is confusing and seemingly unnecessary. We have logic to remove nodes from `readyNodes` if the client is not ready.

Reply:
Sounds good. It does look pretty unnecessary to me too.